src/rest_utils.cpp

```diff
@@ -188,6 +193,248 @@ Status makeJsonFromPredictResponse(
+            if (seekDataInValField) {
+                auto status = checkValField(tensor.contents().uint_contents_size(), expectedElementsNumber);
+                if (!status.ok())
+                    return status;
+                for (auto& number : tensor.contents().uint_contents()) {
+                    tensor_data.PushBack(number, response.GetAllocator());
+                }
+            } else {
+                fillTensorDataWithValuesFromRawContents<uint8_t>(tensor_data, response_proto, tensor_it, response.GetAllocator());
+            }
+        } else if (tensor.datatype() == "FP64") {
+            if (seekDataInValField) {
+                auto status = checkValField(tensor.contents().fp64_contents_size(), expectedElementsNumber);
+                if (!status.ok())
+                    return status;
+                for (auto& number : tensor.contents().fp64_contents()) {
+                    tensor_data.PushBack(number, response.GetAllocator());
+                }
+            } else {
+                fillTensorDataWithValuesFromRawContents<double>(tensor_data, response_proto, tensor_it, response.GetAllocator());
+            }
+        } else {
+            return StatusCode::REST_UNSUPPORTED_PRECISION;
+        }
+
+        output.AddMember("data", tensor_data, response.GetAllocator());
+        outputs.PushBack(output, response.GetAllocator());
+        tensor_it++;
+    }
+
+    response.AddMember("outputs", outputs, response.GetAllocator());
+    return StatusCode::OK;
+}
+
+Status makeJsonFromPredictResponse(
+    const ::inference::ModelInferResponse& response_proto,
+    std::string* response_json) {
+    Timer timer;
+    using std::chrono::microseconds;
+    timer.start("convert");
+
+    rapidjson::Document response;
+    response.SetObject();
+    rapidjson::Value model_name, id;
+    model_name = rapidjson::StringRef(response_proto.model_name().c_str());
+    id = rapidjson::StringRef(response_proto.id().c_str());
+    response.AddMember("model_name", model_name, response.GetAllocator());
+    response.AddMember("id", id, response.GetAllocator());
+    if (response_proto.model_version().length() > 0) {
+        rapidjson::Value model_version;
+        model_version = rapidjson::StringRef(response_proto.model_version().c_str());
+        response.AddMember("model_version", model_version, response.GetAllocator());
+    }
+
+    auto status = parseParameters(response_proto, response, response.GetAllocator());
+    if(!status.ok())
+    {
+        return status;
+    }
+
+    if(response_proto.outputs_size() == 0){
+        SPDLOG_ERROR("Creating json from tensors failed: No outputs found.");
+        return StatusCode::REST_PROTO_TO_STRING_ERROR;
+    }
+
+    status = parseOutputs(response_proto, response);
+    if(!status.ok())
+    {
+        return status;
+    }
+
+    rapidjson::StringBuffer buffer;
+    rapidjson::PrettyWriter<rapidjson::StringBuffer> writer(buffer);
```

Comment:
You changed this from Writer to PrettyWritter - did you check the performance/message size  impact of this change?